Enformer.pm

```diff
@@ -0,0 +1,220 @@
+ 
+ Please cite the Enformer publication alongside the VEP if you use this resource:
+ https://www.nature.com/articles/s41592-021-01252-x
+
+ Enformer scores can be downloaded from https://ftp.ensembl.org/pub/current_variation/Enformer for GRCh37 and GRCh38.
+
+ The plugin can then be run as default:
+ ./vep -i variations.vcf --assembly GRCh38 --plugin Enformer,file=/path/to/Enformer/data.vcf.gz
+
+ or run with option to only retrieve the SAD (SNP Activity Difference (SAD) scores - main variant effect score computed as model(alternate_sequence) - model(reference_sequence) score 
+ ./vep -i variations.vcf --assembly GRCh38 --plugin Enformer,file=/path/to/Enformer/data.vcf.gz,SAD=1 
+ 
+ or run with option to only retrieve the SAR (Same as SAD, by computing np.log2(1 + model(alternate_sequence)) - np.log2(1 + model(reference_sequence)) score 
+ ./vep -i variations.vcf --assembly GRCh38 --plugin Enformer,file=/path/to/Enformer/data.vcf.gz,SAR=1 
+
+ or run with option to only retrieve the principal component scores which are a reduced representation of a much bigger vector of the SAD and SAR after using principal component analysis (PCA)
+ ./vep -i variations.vcf --assembly GRCh38 --plugin Enformer,file=/path/to/Enformer/data.vcf.gz,PC=1 
+
+
+ The tabix utility must be installed in your path to use this plugin.
+ Check https://github.com/samtools/htslib.git for instructions.
+
+
+=cut
+
+package Enformer;
+
+use strict;
+use warnings; 
+
+use Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin;
+use Bio::EnsEMBL::Variation::Utils::Sequence qw(get_matched_variant_alleles);
+use base qw(Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin);
+
+sub new {
+  my $class = shift;
+  
+  my $self = $class->SUPER::new(@_);
+
+  $self->expand_left(0);
+  $self->expand_right(0);
+  $self->get_user_params();
+  
+  my $params = $self->params_to_hash();
+  my $file;
+   
+  die "File needs to be specified to run the Enformer plugin. File can be downloaded from  https://ftp.ensembl.org/pub/current_variation/Enformer \n" if  (!%$params);
+  if (!keys %$params) {
+    $file = $self->params->[0];
+    $params->{file} = $file;
+  } else {
+    $file = $params->{file};
+    $self->{SAD} = $params->{SAD} if (defined ($params->{SAD}));
+    $self->{SAR} = $params->{SAR} if (defined ($params->{SAR}));
+    $self->{PC}  = $params->{PC} if (defined ($params->{PC}));
+  } 
+
+  delete $params->{SAD} if defined $params->{SAD} && $params->{SAD} eq '0';
+  delete $params->{SAR} if defined $params->{SAR} && $params->{SAR} eq '0';
+  delete $params->{PC} if defined $params->{PC} && $params->{PC} eq '0';
+  
+  $self->{params} = $params;
+  $self->add_file($file);
+  
+
+  return $self;
+
+}
+
+sub feature_types {
+  return ['Transcript', 'MotifFeature', 'Intergenic'];
+}
+
+sub get_header_info {
+  my $self = shift;
+
+  my %header; 
+
+  if (keys(%{$self->{params}}) == 1)  {
+    $header{"Enformer"} = "Prediction tool to accurately identify variant impact on gene expression";
```

Comment:
We don't return a value for 'Enformer' so don't need a header; we need something about predicting variant impact on gene expression in the headers that are used.